rust/test_contract_2/src/lib.rs

```diff
@@ -95,6 +132,36 @@ mod service {
         async fn event(&self, id: u64) -> Result<event_structs::HistoryEvents, anyhow::Error> {
             get_event(id)
         }
+
+        async fn get_add_events(
```

Comment:
Nothing using `condition`?

Reply:
I had tested it manually before... but you're right. I just added a bunch of tests to demonstrate that the other functionality is working as expected